openff/benchmark/geometry_optimizations/compute.py

```diff
@@ -66,21 +69,62 @@ def submit_molecules(server_uri, input_paths, season, dataset_name="Benchmark Op
     logging.info("Submitted!")
 
 
-def export_molecule_data(server_uri, destination_path):
-    """Export molecule data from target QCFractal instance.
+def export_molecule_data(server_uri, output_directory, dataset_name="Benchmark Optimizations"):
+    """Export all molecule data from target QCFractal server to the given directory.
 
     Parameters
     ----------
-
+    server_uri : str
+        Target QCFractal server URI.
+    destination_path : str
+        Directory path to deposit exported data.
+    dataset_name : str
+        Dataset name to extract from the QCFractal server.
 
     """
-    # export all molecule/optimization data from all datasets 
-
-    # SDF key-value pairs should be used for method, basis, program, provenance, `openff-benchmark` version
-
-    # SDF key-value pairs also used for final energies
-
-    # subfolders for each compute spec, files named according to molecule ids
+    # get dataset
+    client = FractalClient(server_uri, verify=False)
+    optds = client.get_collection("OptimizationDataset", dataset_name)
+    optds.status()
+
+    try:
+        os.makedirs(output_directory)
+    except OSError:
+        raise Exception(f'Output directory {output_directory} already exists. '
+                         'The user must delete this manually.')
+
+    # for each compute spec, create a folder in the output directory
+    # deposit SDF giving final molecule, energy
+    specs = optds.list_specifications().index.tolist()
+    for spec in specs:
+        os.makedirs(os.path.join(output_directory, spec))
+        optentspec = optds.get_specification(spec)
+
+        for id, opt in optds.df[spec].iteritems():
+            mol = Molecule.from_qcschema(
```

Comment:
Nice I forgot about this method, but this is only pulling out the [initial confirmation](https://github.com/openforcefield/openforcefield/blob/1183ea03102212a5da3df092a359bc63990b98c0/openforcefield/topology/molecule.py#L4584) you need to manually add the final conformation so its probably faster to make the molecule using the method and not pass the client then query the final molecule, then you don't have a double query.

Reply:
Fixed! Great catch @jthorton!